README.md

```diff
@@ -391,6 +391,55 @@ property limits styles to highlighted languages.
 When converting a Prism CSS theme it's mostly just necessary to use classes as
 `types` and convert the declarations to object-style-syntax and put them on `style`.
 
+### SSR Support / Avoiding FOUC
+
+If your React app supports "light mode / dark mode" you need to do additional work to avoid a flash of unstyled content (FOUC). Generate the following script tag and inject it into your HTML so it runs _before_ your content loads.
+
+```js
+import { generateScriptForSSR } from 'prism-react-renderer'
+import duotoneDark from 'prism-react-renderer/themes/duotoneDark';
+import duotoneLight from 'prism-react-renderer/themes/duotoneLight';
+
+// A stringified function returning the `id` of the
+// theme you wish to render on the initial page load
+const getThemeIdFuncStr = `
+  () => (
+    window.localStorage.get('color-mode') === 'dark'
+    ? '${duotoneDark.id}'
+    : '${duotoneLight.id}'
+  );
+`.trim()
+
+const codeToRunOnClient = generateScriptForSSR(
+  // Include whatever themes `getThemeIdFuncStr` might return
+  [duotoneDark, duotoneLight],
+  getThemeIdFuncStr
+);
+
+// Gatsby
+export const onRenderBody = ({ setPreBodyComponents }) => {
+  setPreBodyComponents(
+    <script dangerouslySetInnerHTML={{ __html: codeToRunOnClient }} />
+  );

```

Comment:
Not sure if we have any way around this, but if end-user is using any sort of Content Security Policy header to, say, restrict inline scripts – then this will likely fail. I don't think very many end-users would be doing that, but just wanted to point that out as a potential failure point here.

Reply:
Ah yes, I totally didn't think about that. I'll add a warning about it and a way to work with CSP.

Reply:
Added a note about CSP.

Reply:
Should we sanitize `codeToRunOnClient`? I see that we are sometimes getting theme values from localStorage, which means the theme data isn't guaranteed to be as expected. 

Forgive me if this comment is naive or intrusive. I am in the middle of security training which is why the dangers of `dangerouslySetInnerHTML` are at the front of my mind. :)

Reply:
Hi @ZimChristine, thanks for the question! I _think_ this code is fine because I believe the worst thing that could happen is we set the wrong CSS variables, but I would love to have a second opinion!

Reply:
I'll chime in here. I think this is a very valid question! I think in this case, we'll be okay – since the code that we're "dangerously" setting is generated from the site _developer_ and not from _users_. Generally, injection attacks occur when you fail to sanitize user-provided data. In this case, the code we're "injecting" here is provided explicitly by the site developer, and is not really dynamic – so if there were some sort of _shenanigans_ going on, it'd be coming from the site developer themselves, and there's not much we can do about that.

Reply:
Right on! Thanks for the reply. Agreed that for the current usage this dangerously set html is protected from an injection attack; glad for the dialogue. :)